src/components/icon/index.tsx

```diff
@@ -37,10 +37,15 @@ export type IconProps = ImageProps &
 
 type Props = IconProps & BaseComponentInjectedProps;
 
-const Icon = (props: Props) => {
+const defaultWebIconSize = 16;
+
+const Icon = forwardRef((props: Props, ref: any) => {
   const {size, tintColor, style, supportRTL, source, assetGroup, assetName, modifiers, ...others} = props;
   const {margins} = modifiers;
-  const iconSize = size ? {width: size, height: size} : undefined;
+  const webIconSize = Constants.isWeb
+    ? {width: size || defaultWebIconSize, height: size || defaultWebIconSize}
+    : undefined;
+  const iconSize = size ? {width: size, height: size} : webIconSize;
```

Comment:
I suggest encapsulating this logic under a hook for useIconSize. It will handle all the logic and return the final size. 
Right now it's getting a little unreadable

Reply:
See if you can do something like that when you destruct the size prop from props with a default prop
```
const {
  size = Constants.isWeb ? defaultWebIconSize : undefined,
  tintColor,
  style,
  .....
}

```